API/utils/util.go

```diff
@@ -349,55 +350,82 @@ func GetEntitiesByNamespace(namespace Namespace, hierarchyName string) []string
 	case PStray:
 		entNames = append(entNames, EntityToString(STRAYOBJ))
 	case Physical, PHierarchy, Any:
-		if hierarchyName == "" {
+		entities := []int{}
+
+		if hierarchyName == "" || hierarchyName == "**" {
 			// All entities of each namespace
 			switch namespace {
 			case Physical:
-				for i := STRAYOBJ; i <= GROUP; i++ {
-					entNames = append(entNames, EntityToString(i))
+				for entity := STRAYOBJ; entity <= GROUP; entity++ {
+					entities = append(entities, entity)
 				}
 			case PHierarchy:
-				for i := SITE; i <= GROUP; i++ {
-					entNames = append(entNames, EntityToString(i))
+				for entity := SITE; entity <= GROUP; entity++ {
+					entities = append(entities, entity)
 				}
 			case Any:
-				// All collections
-				for _, entity := range Entities {
-					entNames = append(entNames, EntityToString(entity))
-				}
+				entities = Entities
 			}
 		} else {
-			// Add entities according to hierarchyName possibilities
-			resp := []int{}
 			if namespace == Any {
-				resp = append(resp, DOMAIN)
+				entities = append(entities, DOMAIN)
 			}
-			switch strings.Count(hierarchyName, HN_DELIMETER) {
-			case 0:
-				resp = append(resp, SITE)
-				if namespace == Any {
-					resp = append(resp, OBJTMPL, ROOMTMPL, BLDGTMPL, TAG, LAYER)
+
+			// Add entities according to hierarchyName possibilities
+			if strings.Contains(hierarchyName, ".**") {
+				var initialEntity int
+				finalEntity := GROUP
+
+				switch strings.Count(hierarchyName, HN_DELIMETER) {
+				case 1:
+					initialEntity = SITE
```

Comment:
it's searching its own entity unnecessarily. For example, siteA.** will also search site but it cannot have a children of siteA.

Reply:
The same thing applies to others after too, rooms and racks.